tests/integration/targets/vm_snapshot/tasks/01_vm_snapshot_info.yml

```diff
@@ -0,0 +1,172 @@
+---
+- name: Test vm_snapshot_info
+  vars:
+    nonexistent_snapshot_label: no-label
+    nonexistent_snapshot_vm_name: no-vm-name
+    nonexistent_snapshot_serial: -1
+    non_unique_snapshot_label: not-unique
+    created_snapshot_test_vm_1:
+      vm_name: snapshot-test-vm-1
+      snap_labels:
+        - snap-0
+        - snap-1
+        - snap-2
+    created_snapshot_test_vm_2:
+      vm_name: snapshot-test-vm-2
+      snap_labels:
+        - "{{ non_unique_snapshot_label }}"
+  block:
+    - include_tasks: helper_api_vm_snapshot_create.yml
+      vars:
+        vms_number: "{{ test_vms_number }}"
+
+    # --------------------------------------------------------
+
+    - name: List all VM snapshots - API
+      scale_computing.hypercore.api:
+        action: get
+        endpoint: /rest/v1/VirDomainSnapshot
+      register: vm_snapshots_api
+    - ansible.builtin.set_fact:
+        existing_snapshot_label: "{{ vm_snapshots_api.record[0].label }}"
```

Comment:
You do not know what is at [0] -  vm_snapshots_api.record[0]. If test passes - ok, I guess you are lucky. But you need to check such assumptions, before you continue.